x-pack/platform/plugins/shared/alerting/server/rules_client/common/inject_references.ts

```diff
@@ -77,3 +77,26 @@ export function injectReferencesIntoParams<
     );
   }
 }
+
+export function injectReferencesIntoArtifacts(
+  ruleId: string,
+  artifacts: RawRule['artifacts'],
+  references: SavedObjectReference[]
+) {
+  if (!artifacts) {
+    return { dashboards: [] };
+  }
+  return {
+    ...artifacts,
+    dashboards: artifacts.dashboards?.map((dashboard) => {
+      const reference = references.find((ref) => ref.name === dashboard.refId);
```

Comment:
```suggestion
      const reference = references.find((ref) => ref.name === dashboard.refId && ref.type === 'dashboard');
```

I am wondering if we should be cautious and make the above suggestion. I'm unsure; maybe the `dashboard_` prefix is enough, and the refs (actions and dashboards) will never get messed up.